### PR TITLE
[3/n] Refactor workflow details page to use v2 nodes route

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR refactors workflow details page to use v2 nodes route. We mainly handle node selection and retrieval logic to directly use `nodes` from V2 endpoints, rather than the convoluted `operators` and `artifacts` objects in `selectedDag` reducer state.

We are not yet touching the node rendering part, which is a huge change and will include in a separate PR.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/pages/workflow/id/index.tsx [hunk 1]
```diff
@@ -94,18 +100,24 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
   const [showUpdateMessage, setShowUpdateMessage] = useState<boolean>(false);
   const [updateSucceeded, setUpdateSucceeded] = useState<boolean>(false);
 
-  const currentNode = useSelector(
-    (state: RootState) => state.nodeSelectionReducer.selected
+  const selectedNodeState = useSelector(
+    (state: RootState) =>
+      state.workflowPageReducer.perWorkflowPageStates[workflowId]?.SelectedNode
```
Comment (agiron123) [inline src/ui/common/src/components/pages/workflow/id/index.tsx:105]: what does this perWorkflowPageStates do? Does it manage the sidesheet state?

Comment (likawind) [reply src/ui/common/src/components/pages/workflow/id/index.tsx:105]: After getting deep into most refactors, I think the only thing it's managing is the selected node (which controls whether to show and which sidesheet to show). It doesn't further manage any page states, which will be controlled by page components or already handled by RTK

**Review by agiron123 — APPROVED** [review]

File: src/golang/lib/response/node.go [hunk 2]
```diff
@@ -120,6 +120,7 @@ func NewArtifactFromDBObject(dbArtifactNode *views.ArtifactNode) *Artifact {
 type ArtifactResult struct {
 	// Contains only the `result`. It mostly mirrors 'artifact_result' schema.
 	ID                uuid.UUID                        `json:"id"`
+	ArtifactID        uuid.UUID                        `json:"artifact_id"`
```
Comment (likawind) [inline src/golang/lib/response/node.go:123]: @eunice-chan

File: src/ui/common/src/components/pages/workflow/id/index.tsx (hunk 1 above)
Comment (agiron123) [reply src/ui/common/src/components/pages/workflow/id/index.tsx:105]: sounds good

**Review by agiron123 — APPROVED** [review]

**Pull request merged**